TestHttpsInsecure failing on Plan 9

Changeset 696c91bc6dcd6c603a1685c5f623f5ca0ddafc06 added TestHttpsInsecure, which is failing on Plan 9:

```
cpu% go test -v -run TestHttpsInsecure
=== RUN   TestHttpsInsecure
--- FAIL: TestHttpsInsecure (11.28s)
	fetch_test.go:420: fetchProfiles(https+insecure://127.0.0.1:47334/debug/pprof/profile) got non-symbolized profile: len(p.Function)==0
	fetch_test.go:376: Serve got error accept tcp 127.0.0.1:47334: open /net/tcp/11/listen: '/net/tcp/11/listen' listen hungup, want "use of closed"
FAIL
exit status: 'driver.test 1773648: 1'
FAIL	cmd/vendor/github.com/google/pprof/internal/driver	11.370s
```

There are two issues:

- CPU profiling is not implemented on Plan 9, so the profile is empty.
- Closing a closed listener on Plan 9 returns `listen hungup` instead of `use of closed` (see [/sys/src/9/ip/devip.c:476](https://github.com/0intro/plan9/blob/4225c3e/sys/src/9/ip/devip.c#L476)).

--- a/internal/driver/fetch_test.go
+++ b/internal/driver/fetch_test.go
@@ -351,6 +351,13 @@ func stubHTTPGet(source string, _ time.Duration) (*http.Response, error) {
 	return c.Get("file:///" + file)
 }
 
+func closedError() string {
+	if runtime.GOOS == "plan9" {
+		return "listen hungup"
+	}
+	return "use of closed"
+}
+
 func TestHttpsInsecure(t *testing.T) {
 	if runtime.GOOS == "nacl" {
 		t.Skip("test assumes tcp available")
@@ -372,7 +379,7 @@ func TestHttpsInsecure(t *testing.T) {
 		donec <- http.Serve(l, nil)
 	}(donec)
 	defer func() {
-		if got, want := <-donec, "use of closed"; !strings.Contains(got.Error(), want) {
+		if got, want := <-donec, closedError(); !strings.Contains(got.Error(), want) {
 			t.Fatalf("Serve got error %v, want %q", got, want)
 		}
 	}()

--- a/internal/driver/fetch_test.go
+++ b/internal/driver/fetch_test.go
@@ -423,6 +423,10 @@ func TestHttpsInsecure(t *testing.T) {
 	if len(p.SampleType) == 0 {
 		t.Fatalf("fetchProfiles(%s) got empty profile: len(p.SampleType)==0", address)
 	}
+	if runtime.GOOS == "plan9" {
+		// CPU profiling is not supported on Plan9; see golang.org/issues/22564.
+		return
+	}
 	if len(p.Function) == 0 {
 		t.Fatalf("fetchProfiles(%s) got non-symbolized profile: len(p.Function)==0", address)
 	}
@@ -436,7 +440,6 @@ func TestHttpsInsecure(t *testing.T) {
 var badSigprofOS = map[string]bool{
 	"darwin": true,
 	"netbsd": true,
-	"plan9":  true,
 }
 
 func checkProfileHasFunction(p *profile.Profile, fname string) error {

